src/Backend/SgateShopgatePlugin/Controllers/Frontend/shopgate.php

```diff
@@ -1006,6 +1281,654 @@ public function registrationAction()
+    protected function addArticleToWishList($articles, $userId)
+    {
+        foreach ($articles as $article) {
+            $articleId = trim($article['product_id']);
+            $orderNumber = trim($article['variant_id']);
+
+            $product = Shopware()->Modules()->Articles()->sGetArticleById($articleId);
+
+            if ($product) {
+                if ($orderNumber === "") {
+                    $orderNumber = $product['ordernumber'];
+                }
+
+                if ($this->isWishList()) {
+
+                    $builder = $this->getModelManager()->createQueryBuilder();
+                    $builder->select(array('cart', 'items', 'details'))
+                        ->from('SwagAdvancedCart\Models\Cart\Cart', 'cart')
+                        ->leftJoin('cart.customer', 'customer')
+                        ->leftJoin('cart.cartItems', 'items')
+                        ->leftJoin('items.details', 'details')
+                        ->where('customer.id = :customerId')
+                        ->andWhere('cart.shopId = :shopId')
+                        ->andWhere('LENGTH(cart.name) > 0')
+                        ->orderBy('items.id', 'DESC')
+                        ->setParameter('customerId', $userId)
+                        ->setParameter('shopId', $this->get('shop')->getId());
+
+                    $carts = $builder->getQuery()->getResult();
+
+                    if (empty($carts)) {
+                        $this->Request()->setPost('newlist', 'App');
+                    } else {
+                        $this->Request()->setPost('lists', array($carts[0]->getId()));
+                    }
+
+                    $this->Request()->setPost('ordernumber', $orderNumber);
+                    $this->container->get('swag_advanced_cart.cart_handler')->addToList($this->Request()->getPost());
+                } else {
+                    $productId = Shopware()->Modules()->Articles()->sGetArticleIdByOrderNumber($orderNumber);
+                    $productName = Shopware()->Modules()->Articles()->sGetArticleNameByOrderNumber($orderNumber);
+
+                    if (empty($productId)) {
+                        return false;
+                    }
+
+                    Shopware()->Modules()->Basket()->sAddNote($productId, $productName, $orderNumber);
+                }
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * A helper function to check if the wish list from the advanced cart plugin is active
+     *
+     * @return bool
+     */
+    private function isWishList()
+    {
+        $pluginManager  = $this->container->get('shopware_plugininstaller.plugin_manager');
+        $plugin = $pluginManager->getPluginByName('SwagAdvancedCart');
+        $config = $this->container->get('shopware.plugin.cached_config_reader')->getByPluginName('SwagAdvancedCart');
+
+        if ($plugin->getInstalled() && $plugin->getActive() && $config['replaceNote']) {
```

Comment:
this can just be `return $plugin->getInstalled() && $plugin->getActive() && $config['replaceNote'];`
then you don't need the `true|false` returns